### DISCUSSION TITLE
Install from api: is it safe to delete local core/cask git repo?

### DISCUSSION THREAD

**Discussion by daviehh** [Homebrew/discussions#2365] [Everyday usage] [answered]
Following the recent 3.3.0 [update](https://brew.sh/2021/10/25/homebrew-3.3.0/), I'm interested in trying out the `HOMEBREW_INSTALL_FROM_API` flag. After setting it in .zshrc, is it now safe to remove the local homebrew core and cask repos (e.g. `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core`) to save disk space? Thanks!

**Comment by Rylan12** [comment, marked as answer]
Yep! This is exactly why we created `HOMEBREW_INSTALL_FROM_API`!

You can uninstall by running `brew untap homebrew/core` and `brew untap homebrew/cask`.

Please note that this is still a very new feature so there's a chance that some minor issues may occur. If you run into anything weird or have any feedback, please let us know by opening an issue or discussion.

**Comment by grantbarrett** [comment]
When I uninstall, I get the message `Refusing to untap homebrew/core because it contains the following installed formulae or casks` with a list. Is there a way to indicate that those items should now be affiliated with the API?

**Reply by Rylan12** [reply]
Thanks for the report! This will be fixed by https://github.com/Homebrew/brew/pull/12346

**Reply by tlk** [reply]
@Rylan12 This looks very promising, much appreciated!

I just experimented with it and it looks like its working :-)

```
% brew update; brew upgrade
% brew developer on; brew update; brew upgrade; brew developer off   # upgrade brew command to 3.3.1-47-gdae9a34
% HOMEBREW_INSTALL_FROM_API=1 brew untap homebrew/cask
% HOMEBREW_INSTALL_FROM_API=1 brew untap homebrew/core
% HOMEBREW_INSTALL_FROM_API=1 brew cleanup
% HOMEBREW_INSTALL_FROM_API=1 brew autoremove
% HOMEBREW_INSTALL_FROM_API=1 brew install hello
% HOMEBREW_INSTALL_FROM_API=1 brew uninstall hello
```

All the above appeared to work out well, assuming my guesswork makes sense.

I did notice something surprising, the spotify cask is being upgraded every time I run `brew upgrade`. That didn't happen earlier, I'm pretty sure. Not sure if that is related?

```
% brew upgrade
==> Casks with 'auto_updates true' or 'version :latest' will not be upgraded; pass `--greedy` to upgrade them.
==> Upgrading 1 outdated package:
spotify latest -> latest
==> Upgrading spotify
 ..
```

**Reply by Rylan12** [reply]
Thanks, @tlk. If you don't mind, can you report this as a bug by opening an issue in Homebrew/brew?

We were having some similar issues a while back but I thought I had fixed it. Either way, someone else reported a similar issue so I think you're not alone. I'm looking into it but it may take some time to solve.

**Comment by Rylan12** [comment]
I just released Homebrew 3.3.2 which contains fixes for the original problem. Now, `brew untap homebrew/core` should work as expected, even if you have formulae installed already.

The release also contains a handful of other fixes so hopefully, some other issues that people were encountering won't occur anymore. As always, feel free to keep reporting new issues. If it's a bug, feel free to go ahead and report directly to Homebrew/brew via GitHub issues.

**Reply by grantbarrett** [reply]
Thank you for this fix.
